sdmetrics/multi_table/statistical/cardinality_shape_similarity.py

```diff
@@ -0,0 +1,127 @@
+"""The CardinalityShapeSimilarity metric."""
+
+import numpy as np
+from scipy.stats import ks_2samp
+
+from sdmetrics.goal import Goal
+from sdmetrics.multi_table.base import MultiTableMetric
+from sdmetrics.utils import get_cardinality_distribution
+
+
+class CardinalityShapeSimilarity(MultiTableMetric):
+    """CardinalityShapeSimilarity metric computes the similarity of the cardinality.
+
+    Attributes:
+        name (str):
+            Name to use when reports about this metric are printed.
+        goal (sdmetrics.goal.Goal):
+            The goal of this metric.
+        min_value (float):
+            Minimum value that this metric can take.
+        max_value (float):
+            Maximum value that this metric can take.
+    """
+
+    name = 'CardinalityShapeSimilarity'
+    goal = Goal.MAXIMIZE
+    min_value = 0.0
+    max_value = 1.0
+
+    @classmethod
+    def compute_breakdown(cls, real_data, synthetic_data, metadata=None):
+        """Compute the breakdown of cardinality shape similarity in the given tables.
+
+        Compute the cardinality distributions for the real and synthetic data for each
+        (parent, child) relationship specified in the metadata. Then compute the KSComplement
+        over the two cardinality distributions, and compare the final scores
+        to obtain the cardinality shape similarity score.
+
+        Args:
+            real_data (dict[str, pandas.DataFrame]):
+                The tables from the real dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            synthetic_data (dict[str, pandas.DataFrame]):
+                The tables from the synthetic dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            metadata (dict):
+                Multi-table metadata dict. If not passed, it is build based on the
+                real_data fields and dtypes.
+
+        Returns:
+            dict:
+                A dict mapping (parent, child) values to the score breakdown for the
+                cardinality distributions of that foreign key.
+        """
+        if set(real_data.keys()) != set(synthetic_data.keys()):
+            raise ValueError('`real_data` and `synthetic_data` must have the same tables.')
+        if metadata is None:
+            raise ValueError('`metadata` cannot be ``None``.')
```

Comment:
If we require `metadata` then change the `metadata=None` to just `metadata`, that way it must always be passed. Then we don't need to do this check.

Reply:
Hm yeah we could do that, but I was wondering if it's confusing since the parent method we are overriding has `metadata` as optional. I guess it's fine since we are effectively enforcing this rule anyway.